0-bootstrap/modules/groups/variables.tf

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "create_groups_holder" {
+  description = "Contain the details of the Groups to be created"
+  type = object({
+    create_groups = bool
```

Comment:
Do we need this var or can we statically determine this if any value != ""?

Reply:
It's helping to keep the code simpler during the usage of the variable and at the validation. So if the user wants to make use of the automatic groups creation, he just set this value to true and the flow will follow this approach.

There's also a difference between the required and optional groups to be created, the required must be all filled or not, while for the optional we don't necessary want to create all of them by just passing their values as empty. With this variable we can check this validation easier from inside the variable definition.

Reply:
Sounds good!